Add progress reporting.

psoo opened the pull request:
This optionally prints the progress of pg_checksums
via bytes read/written to the terminal with the new
command line argument -P.

psoo committed:
Add progress reporting.

This optionally prints the progress of pg_checksums
via bytes read/written to the terminal with the new
command line argument -P.

psoo committed:
Enable progress status reporting to be enabled during runtime.

If -P was forgotten and pg_checksums operates on a large cluster,
the caller can send SIGUSR1 to pg_checksums to turn progress status
reporting on during runtime.

mbanck commented on pg_checksums.c:
```diff
@@ -117,6 +118,17 @@ static const char *skip[] = {
 
+static void enable_progress_report(int signo,
+								   siginfo_t *siginfo,
+								   void *context) {
```
I'd prefer if we keep using the PostgreSQL style indentation here, i.e. opening braces of functions on a new line

mbanck commented on pg_checksums.c:
```diff
@@ -117,6 +118,17 @@ static const char *skip[] = {
+								   void *context) {
+
+	/* we handle SIGUSR1 only */
+	if (signo == SIGUSR1) {
```
On the other hand, the braces could be removed in this one.

mbanck commented on pg_checksums.c:
```diff
@@ -784,6 +798,24 @@ main(int argc, char *argv[])
+		 */
+		memset(&act, '\0', sizeof(act));
+		act.sa_sigaction = &enable_progress_report;
+		act.sa_flags  = SA_SIGINFO;
```
two blanks before the =

mbanck commented on pg_checksums.c:
```diff
@@ -784,6 +798,24 @@ main(int argc, char *argv[])
+		 * message on STDERR.
+		 */
+		if (sigaction(SIGUSR1, &act, NULL) < 0) {
+			fprintf(stderr, "could not set signal handler to turn on progress bar\n");
```
Standard format for error messages is pfrintf(stderr, _("%s: foo\n), progname);

mbanck commented on pg_checksums.c:
```diff
@@ -286,17 +347,21 @@ scan_file(char *fn, int segmentno)
-	DIR		   *dir;
+	int64          dirsize = 0;
+	char		   path[MAXPGPATH];
+	DIR		      *dir;
```
dirsize is new, but the other two look like unrelated whitespace changes, did you check that those are legit?

psoo committed:
Cleanup coding style issues and run pg_indent.

This should repair some of the indentation issues found by
Michael Banck during his review and make the code
look like described in the PostgreSQL styleguide.

psoo committed:
Teach report_progress() to print the current r/w speed, too.

psoo committed:
Another run of pgindent.
